examples/nature_of_code/chp_03_oscillation/3_10_exercise_oop_wave.rs

```diff
@@ -0,0 +1,109 @@
+// The Nature of Code
+// Daniel Shiffman
+// http://natureofcode.com
+//
+// Exercise 3-10: OOP Wave
+extern crate nannou;
+
+use nannou::prelude::*;
+
+fn main() {
+    nannou::app(model, event, view).run();
+}
+
+struct Model {
+    wave0: Wave,
+    wave1: Wave,
+}
+
+struct Wave {
+    x_spacing: f32,       // How far apart should each horizontal position be spaced
+    _w: f32,              // Width of entire wave
+    origin: Vector2<f32>, // Where does the wave's first point start
+    theta: f32,           // Start angle at 0
+    amplitude: f32,       // Height of the wave
+    _period: f32,         // How many pixels before the wave repeats
+    dx: f32, // Value for incementing X, to be calculated as a function of period and x_spacing
+    y_values: Vec<f32>, // Using a vector to store the height values for the wave (not entirely necessary)
+}
+
+impl Wave {
+    fn new(o: Vector2<f32>, w: f32, a: f32, p: f32) -> Self {
+        let origin = o;
+        let x_spacing = 8.0 as f32;
+        let theta = 0.0 as f32;
+        let _w = w;
+        let _period = p;
+        let amplitude = a;
+        let dx = (((std::f32::consts::PI * 2.0) / _period) * x_spacing) as f32;
```

Comment:
`PI` should be imported automatically via prelude now, so shouldn't need `std::f32::consts::`.